zio-http/src/main/scala/zhttp/service/Server.scala

```diff
@@ -36,6 +36,104 @@ sealed trait Server[-R, +E] { self =>
+   */
+  def withBind(hostname: String, port: Int): Server[R, E] =
+    Concat(self, Server.Address(new InetSocketAddress(hostname, port)))
+
+  /**
+   * Creates a new server listening on the provided InetAddress and port.
+   */
+  def withBind(address: InetAddress, port: Int): Server[R, E] =
+    Concat(self, Server.Address(new InetSocketAddress(address, port)))
+
+  /**
+   * Creates a new server listening on the provided InetSocketAddress.
+   */
+  def withBind(inetSocketAddress: InetSocketAddress): Server[R, E] = Concat(self, Server.Address(inetSocketAddress))
+
+  /**
+   * Creates a new server with the errorHandler provided.
+   */
+  def withError[R1](errorHandler: Throwable => ZIO[R1, Nothing, Unit]): Server[R with R1, E] =
+    Concat(self, Server.Error(errorHandler))
+
+  /**
+   * Creates a new server with the following ssl options.
+   */
+  def withSsl(sslOptions: ServerSSLOptions): Server[R, E] = Concat(self, Server.Ssl(sslOptions))
+
+  /**
+   * Creates a new server using a HttpServerExpectContinueHandler to send a 100 HttpResponse if necessary.
+   */
+  def withAcceptContinue: Server[R, E] = Concat(self, Server.AcceptContinue)
+
+  /**
+   * Creates a new server using netty FlowControlHandler (@see <a
+   * href="https://netty.io/4.1/api/io/netty/handler/flow/FlowControlHandler.html">FlowControlHandler</a>).
+   */
+  def withFlowControl: Server[R, E] = Concat(self, Server.FlowControl)
+
+  /**
+   * Creates a new server with leak detection disabled (@see <a
+   * href="https://netty.io/4.1/api/io/netty/util/ResourceLeakDetector.Level.html#DISABLED">Disabled Level</a>).
+   */
+  def withLeakDetectionDisabled: Server[R, E] = Concat(self, LeakDetection(LeakDetectionLevel.DISABLED))
+
+  /**
+   * Creates a new server with simple leak detection (@see <a
+   * href="https://netty.io/4.1/api/io/netty/util/ResourceLeakDetector.Level.html#SIMPLE">Simple Level</a>).
+   */
+  def withSimpleLeakDetection: Server[R, E] = Concat(self, LeakDetection(LeakDetectionLevel.SIMPLE))
+
+  /**
+   * Creates a new server with advanced leak detection (@see <a
+   * href="https://netty.io/4.1/api/io/netty/util/ResourceLeakDetector.Level.html#ADVANCED">Advanced Level</a>).
+   */
+  def withAdvancedLeakDetection: Server[R, E] = Concat(self, LeakDetection(LeakDetectionLevel.ADVANCED))
+
+  /**
+   * Creates a new server with paranoid leak detection (@see <a
+   * href="https://netty.io/4.1/api/io/netty/util/ResourceLeakDetector.Level.html#PARANOID">Paranoid Level</a>).
+   */
+  def withParanoidLeakDetection: Server[R, E] = Concat(self, LeakDetection(LeakDetectionLevel.PARANOID))
+
+  /**
+   * Creates a new server with the leak detection level provided (@see <a
+   * href="https://netty.io/4.1/api/io/netty/util/ResourceLeakDetector.Level.html">ResourceLeakDetector.Level</a>).
+   */
+  def withLeakDetection(level: LeakDetectionLevel): Server[R, E] = Concat(self, LeakDetection(level))
```

Comment:
I think we should just keep this method, and drop the specific ones. IMO, it is just noise.

Reply:
Ok, changed.